### PR TITLE
Extend documentation to guide users through Westeros tutorials

### LINKED ISSUE #475: Extend documentation to guide users through Westeros tutorials
After having removed the numbering of all Westeros tutorials in PR #450 , some guidance is required for users so that they:
a. know where to start
b. know which set of tutorials belong together
c. understand in what order to do tutorials e.g. which are basics and which are more advanced.

### PR DESCRIPTION
**Pull request by LauWien**
This PR extends the documentation to guide users through Westeros tutorials.
Closes #475. 

## How to review

- Read the diff and note that the CI checks all pass.
- Build the documentation and look [the tutorial page](https://docs.messageix.org/en/stable/tutorials.html).
- Look at the[ rendered baseline tutorial](https://github.com/LauWien/message_ix/blob/issue/475/tutorial/westeros/westeros_baseline.ipynb) under section "Recommended order for the tutorials".

## PR checklist

- [x] Continuous integration checks all ✅
- [x] ~Add, expand, or~ update documentation.
- [x] Update release notes.

### CODE REVIEW

File: tutorial/README.rst [hunk 1]
```diff
@@ -105,16 +114,25 @@ uses it to illustrate a range of framework features.
       #. Add a renewable-resource supply curve for the wind power plant,
          (:tut:`westeros/westeros_renewable_resource.ipynb`).
 
-   #. Variablity in energy supply and demand by adding sub-annual time steps,
+   #. Variability in energy supply and demand by adding sub-annual time steps,
```
Comment (OFR-IIASA) [inline tutorial/README.rst:117]: Possibly add list of tutorials below. Currently there is only one tutorial, but more will follow.

**Comment by LauWien** [comment]
Thank you for your review @OFR-IIASA, I implemented all your suggestions.

**Comment by LauWien** [comment]
Thank you @khaeru for jumping in - looks way better now!

File: tutorial/README.rst [hunk 2]
```diff
@@ -105,41 +132,49 @@ uses it to illustrate a range of framework features.
...
+      module to ‘report’ results, e.g. do post-processing, plotting, and other
+      calculations (:tut:`westeros/westeros_report.ipynb`).
 
-   #. Explore historical parameters and their usage
-      (:tut:`westeros/westeros_historical_new_capacity.ipynb`).
-        
-#. After the MESSAGE model has solved, use the :mod:`.message_ix.reporting`
-   module to ‘report’ results, e.g. do post-processing, plotting, and other
-   calculations (:tut:`westeros/westeros_report.ipynb`).
+   #. ⭐ Build the baseline scenario using data stored in Excel files to
```
Comment (LauWien) [inline tutorial/README.rst:164]:
```suggestion
   #. Build the baseline scenario using data stored in Excel files to
```

File: tutorial/README.rst [hunk 3]
```diff
@@ -105,41 +132,49 @@ uses it to illustrate a range of framework features.
...
+      populate sets and parameters:
 
+      #. Export data to file and import the data to create a new scenario
```
Comment (LauWien) [inline tutorial/README.rst:167]:
```suggestion
      #. ⭐ Export data to file and import the data to create a new scenario
```

File: tutorial/README.rst (hunk 2 above)
Comment (khaeru) [reply tutorial/README.rst:164]: Yes please go ahead! It's your PR. I'll approve once done.

**Review by khaeru — APPROVED** [review]

**Comment by khaeru** [comment]
Nothing further from my side; please merge if ready.

**Pull request merged**